nakadi/hack.py

```diff
@@ -120,71 +122,44 @@ def __get_partitions_offsets(topic):
-    parameter = request.args.get(parameter_name)
-
-    if parameter is None:
-        if required:
-            return False, ({'detail': 'missing required query parameter "%s"' % parameter_name}, 400)
-        else:
-            return True, default_value
-
-    if not parameter.isdigit():
-        return False, ({'detail': '"%s" query parameter should be an integer number' % parameter_name}, 400)
-    else:
-        return True, int(parameter)
-
-
 @measured('get_events')
 @authenticate
-def get_events(topic, partition):
+def get_events(topic):
 
     # get and check parameters
-    valid, result = __try_get_parameter_as_int('start_from', flask.request, True)
-    if not valid:
-        return result
-    start_from = result
-
     stream_opts = {}
-    valid, result = __try_get_parameter_as_int('batch_limit', flask.request, False, 1)
-    if not valid:
-        return result
-    stream_opts['batch_limit'] = result
-
-    valid, result = __try_get_parameter_as_int('batch_flush_timeout', flask.request, False, 0)
-    if not valid:
-        return result
-    stream_opts['batch_flush_timeout'] = result
-
-    valid, result = __try_get_parameter_as_int('batch_keep_alive_limit', flask.request, False, -1)
-    if not valid:
-        return result
-    stream_opts['batch_keep_alive_limit'] = result
-
-    valid, result = __try_get_parameter_as_int('stream_limit', flask.request, False, 0)
-    if not valid:
-        return result
-    stream_opts['stream_limit'] = result
-
-    valid, result = __try_get_parameter_as_int('stream_timeout', flask.request, False, 0)
-    if not valid:
-        return result
-    stream_opts['stream_timeout'] = result
-
-    # check that partition is integer
-    if not partition.isdigit():
-        return {'detail': '"partition" path parameter should be an integer number'}, 400
-    else:
-        partition = int(partition)
-
-    # check that topic and partition exist
+    try:
+        stream_opts['batch_limit'] = get_int_parameter('batch_limit', flask.request, False, 1)
+        stream_opts['batch_flush_timeout'] = get_int_parameter('batch_flush_timeout', flask.request, False, 0)
+        stream_opts['batch_keep_alive_limit'] = get_int_parameter('batch_keep_alive_limit', flask.request, False, -1)
+        stream_opts['stream_limit'] = get_int_parameter('stream_limit', flask.request, False, 0)
+        stream_opts['stream_timeout'] = get_int_parameter('stream_timeout', flask.request, False, 0)
+    except NotIntegerParameterException as e:
+        return {'detail': '"%s" query parameter should be an integer number' % e.parameter}, 400
+    except RequiredParameterNotFoundException as e:
+        return {'detail': 'missing required query parameter "%s"' % e.parameter}, 400
+
+    # check that topic exists
     if not __topic_exists(topic):
         return {'detail': 'topic not found'}, 404
-    if not __partition_exists(topic, partition):
-        return {'detail': 'partition not found'}, 404
```

Comment:
I cannot find the code that would check if the partitions, that you fetch from `x-nakadi-cursors` header, are valid,